charts/metal3-deploy/0.1.0/charts/baremetal-operator/templates/configmap-ironic.yaml

```diff
@@ -14,12 +14,14 @@ data:
   CACHEURL: "http://{{ $ironicCacheHost }}/images"
   DEPLOY_KERNEL_URL: "http://{{ $ironicBootHost }}/images/ironic-python-agent.kernel"
   DEPLOY_RAMDISK_URL: "http://{{ $ironicBootHost }}/images/ironic-python-agent.initramfs"
-  HTTP_PORT: {{ .httpPort | quote }}
+  HTTP_PORT: {{ .Values.baremetaloperator.httpPort | quote }}
   IRONIC_ENDPOINT: "http://{{ $ironicApiHost }}/v1/"
   IRONIC_INSPECTOR_ENDPOINT: "http://{{ $ironicInspectorHost }}/v1/"
   DHCP_RANGE: {{ .Values.global.dhcpRange }}
   PROVISIONING_INTERFACE: {{ .Values.global.provisioningInterface }}
+  {{- if .Values.global.provisioningIP }}
   PROVISIONING_IP: {{ .Values.global.provisioningIP }}
+  {{- end }}
   IRONIC_FAST_TRACK: "true"
   RESTART_CONTAINER_CERTIFICATE_UPDATED: "false"
   IRONIC_INSECURE: "true"
```

Comment:
Can this be removed?  AFAICS it's not set in the upstream chart and will disable cert validation for BMO->Ironic connections - if not we should raise a follow-up issue to track fixing this.

Reply:
Actually I see that the IRONIC_ENDPOINT is hard-coded `http` so we perhaps need a follow-up issue to fix that - the Sylva charts have a conditional so that becomes https when TLS is enabled (which should be the default)

Reply:
`IRONIC_INSECURE` seems to not be needed and has been removed. 

Regarding the `http` hardcoding, yes we definitely need an issue for that so we can fix it. This will also open another thing that would need fixing and that is making the Metal3 component images work with TLS and non-root users. Currently this has not been tested and will definitely require additional tweaking of the images. 

This is true mainly because of how we mount our certificates and what permissions do these mount directories have. Currently (at least in Sylva) the default storage class that we can use creates PVs of type `HostPath`, which are notorious for not allowing permission changes on the mount FS. This means that anything that is mounted (in terms of directory structure) is root and cannot be changed (AFAIK). Also the NFS storage class that comes with this chart (and can have its FS permissions changed) cannot be deployed in Sylva as they do not allow NFS storage classes to be deployed in their cluster. Once we setup the first iteration of this chart in Sylva, I will create additional issues, so that we can handle this accordingly.